### PR TITLE
Support 4.0 with key/value list

### PR DESCRIPTION
**Pull request by hanachin**
チケット: https://insidemine.agileware.jp/issues/19085

### CODE REVIEW

File: app/controllers/importer_controller.rb [hunk 1]
```diff
@@ -731,13 +734,33 @@ def version_id_for_name!(project, name, add_versions)
...
   end
 
-  def process_multivalue_custom_field(project, add_versions, issue, custom_field, csv_val)
+  def enumeration_id_for_name!(custom_field, name, add_enumerations)
+    unless @enumeration_id_by_name.key?(name)
+      enumeration = custom_field.enumerations.find_by(name: name).try!(:id)
+      if enumeration.nil?
+        if name.try! { |n| n.size > 1 } && add_enumerations
+          max_position = custom_field.enumerations.pluck(:position).max || 0
+          enumeration = CustomFieldEnumeration.create(custom_field: custom_field, name: name, position: max_position + 1).id
```
Comment (fkdkent) [inline app/controllers/importer_controller.rb]: > `position: max_position + 1`

Is this attribute needed? [`CustomFieldEnumeration#set_position`](https://github.com/redmine/redmine/blob/0454ec48a7d4e5cd45fa62177627ca698940fddd/app/models/custom_field_enumeration.rb#L78-L81) sets position before creation by default.

Comment (hanachin) [reply app/controllers/importer_controller.rb]: Thank you! I didn't know that. I'll fix it.

Comment (hanachin) [reply app/controllers/importer_controller.rb]: Fixed https://github.com/agileware-jp/redmine_importer/pull/5/commits/a6ac56038aafa4a29e7f20c99b5e978e8cd5b243 💪

**Review by fkdkent — APPROVED** [review]

**Review by fkdkent — APPROVED** [review]

**Review by fkdkent — APPROVED** [review]

**Review by fkdkent — APPROVED** [review]

**Review by fkdkent — APPROVED** [review]

File: test/functional/importer_controller_test.rb [hunk 2]
```diff
@@ -20,64 +22,164 @@ def setup
...
+    @issue.update!(assigned_to: @user)
+    post :result, params: build_params(update_issue: 'true').tap { |params| params[:fields_map]['assigned_to'] = 'assigned_to' }
+    assert_response :success
+    assert response.body.include?('Warning')
+    @issue.reload
+    assert_equal 'foobar', @issue.subject
+    assert_equal @user, @issue.assigned_to
+  end
+
+  test 'should error when assigned_to user is not assignable' do
```
Comment (fkdkent) [inline test/functional/importer_controller_test.rb]: Will users see an error in this case? Maybe did you mean like this?

```suggestion
  test 'should unset assigned_to when assigned_to user is not assignable' do
```

Comment (hanachin) [reply test/functional/importer_controller_test.rb]: That is correct, thank you 🙏

**Review by fkdkent — APPROVED** [review]

**Review by fkdkent — APPROVED** [review]

**Pull request merged**